src/FormInterface.php

```diff
@@ -4,21 +4,241 @@
+
+    /**
+     * Removes errors for all attributes or a single attribute.
+     *
+     * @param string|null $attribute attribute name. Use null to remove errors for all attributes.
+     */
     public function clearErrors(?string $attribute = null): void;
+
+    /**
+     * Returns the attribute labels.
+     *
+     * Attribute labels are mainly used for display purpose. For example, given an attribute `firstName`, we can
+     * declare a label `First Name` which is more user-friendly and can be displayed to end users.
+     *
+     * By default an attribute label is generated using {@see generateAttributeLabel()}. This method allows you to
+     * explicitly specify attribute labels.
+     *
+     * Note, in order to inherit labels defined in the parent class, a child class needs to merge the parent labels
+     * with child labels using functions such as `array_merge()`.
+     *
+     * @return array attribute labels (name => label)
+     *
+     * {@see generateAttributeLabel()}
+     */
+    public function attributeLabels(): array;
+
+    /**
+     * Returns the text label for the specified attribute.
+     *
+     * @param string $attribute the attribute name.
+     *
+     * @return string the attribute label.
+     *
+     * {@see generateAttributeLabel()}
+     * {@see attributeLabels()}
+     */
+    public function attributeLabel(string $attribute): string;
+
+    /**
+     * Returns the attribute hints.
+     *
+     * Attribute hints are mainly used for display purpose. For example, given an attribute `isPublic`, we can declare
+     * a hint `Whether the post should be visible for not logged in users`, which provides user-friendly description of
+     * the attribute meaning and can be displayed to end users.
+     *
+     * Unlike label hint will not be generated, if its explicit declaration is omitted.
+     *
+     * Note, in order to inherit hints defined in the parent class, a child class needs to merge the parent hints with
+     * child hints using functions such as `array_merge()`.
+     *
+     * @return array attribute hints (name => hint)
+     */
+    public function attributeHints(): array;
+
+    /**
+     * Returns the text hint for the specified attribute.
+     *
+     * @param string $attribute the attribute name.
+     *
+     * @return string the attribute hint.
+     *
+     * {@see attributeHints()}
+     */
+    public function attributeHint(string $attribute): string;
+
+    /**
+     * Returns a value indicating whether the attribute is required.
+     *
+     * This is determined by checking if the attribute is associated with a {@see \Yiisoft\Validator\Rule\Required}
+     * validation rule.
+     *
+     * @param string $attribute attribute name.
+     *
+     * @return bool whether the attribute is required.
+     */
+    public function isAttributeRequired(string $attribute): bool;
+
+    /**
+     * Returns the list of all the attributes, specifying their typehint.
```

Comment:
What is the use case for this method?

Reply:
It allows to verify if an attribute is required or not by the validator rules, automatically adding forms to the required attribute to the options html.

Reply:
Isn't there `isAttributeRequired()` for it?

Reply:
of course you can configure it in the forms, but suppose it has 10 fields, and I already configure the validator rules, and they are all required, the forms automatically add it.